### ISSUE TITLE
Accuracy difference

### ISSUE THREAD

**Issue by kbramhendra** [nvidia-riva/riva-asrlib-decoder#3]
Hey Hi,
            I am trying to use Riva decoder as a substitution for k2 decoder. With riva decoder I am not getting the same accuracy as that of k2 decoder. There are lot more deletions, at least 12%.  Experimented with different hypereparamters like acoustic_scale, max_active_states, but result seems to be not changing much. I have tried with the different topologies as well(eesen, compact), its the same case with all of them. Can you please help in this regard.

**Comment by messiaen** [comment]
@kbramhendra Thanks for the report. I've passed this on to our team internally.

**Comment by galv** [comment]
@kbramhendra can you check if the branch in #3 fixes your issue? I believe it should based on my own internal testing.

Make sure *not* to set `config.online_opts.lattice_postprocessor_opts.word_boundary_rxfilename` to anything other than empty string to disable worda lignment. See here:

https://github.com/nvidia-riva/riva-asrlib-decoder/commit/3aaa0e113b9f0490995cf0a0617f237eb98ccfd8#diff-c80f4904c78bc561ce1235944f91d4847817445e810c4d7a0064453503e0c7f3L160

Basically, word alignment would fail to complete sometimes when the max_expand option was set. The returned lattice would then be missing paths that were in the input path. Sometimes, these missing paths would be the best cost paths, or sometimes not even a single path would be complete by the time the max_expand option had taken effect. This explains why the errors (at least for me) were only deletions and substitutions, not insertions.

**Comment by jinggaizi** [comment]
@kbramhendra hi, i also try to use this decoder tools, how can use it for ctc+tlg
